src/relcast.erl

```diff
@@ -109,10 +109,12 @@
 %% State record
 %%====================================================================
 
--record(state, {
+-record(state,
+         {
           db :: rocksdb:db_handle(),
           module :: atom(),
           modulestate :: any(),
+          old_serialized :: undefined | map() | binary(),
```

Comment:
Is the typespec for this right? I see undefined and binary, but unclear where "map" would come from. I see it being used to stash modulestate somehwere too.. Very confusing

Reply:
we keep a copy of the old "serialized" state around to compare with, so we don't write things that haven't changed.  with this patch that can either be a binary or a map of binaries and other maps.  see `do_serialize` and its calling code for more details.  I'll write up a better description in the PR description.